### ISSUE TITLE
Middleware does not run when concurrentFeatures is enabled

### ISSUE THREAD

**Issue by saurabhguptarock** [vercel/next.js#31803]
### What version of Next.js are you using?

12.0.5-canary.9

### What version of Node.js are you using?

14.18.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Vercel

### Describe the Bug

Middleware does not run and redirects user.

### Expected Behavior

Middleware should redirect to login page.

### To Reproduce

```
module.exports = {
  images: {
    deviceSizes: [640],
    domains: ["avatars.dicebear.com"],
    formats: ["image/avif", "image/webp"],
  },
  createRoot: true,
  swcMinify: false,
  experimental: {
    reactRoot: true,
    concurrentFeatures: true,
    serverComponents: true,
  },
};
```

_middleware.tsx
```
import type { NextRequest } from "next/server";
import { NextResponse } from "next/server";

export const middleware = (req: NextRequest) => {
  const cookie = req.cookies["auth"];

  if (!cookie) {
    return NextResponse.redirect("/login");
  }
};
```

**Comment by KlotzJesse** [comment]
Just won't work in the main ./pages folder. When added on an api route or /api directly, or in an sub-folder of ./pages it runs.

e.g.
./pages/_middleware.ts wont work
./pages/app/_middleware.ts will work
./pages/api/_middleware.ts will work

 Also in 12.0.5-canary.10

**Comment by saurabhguptarock** [comment]
> Just won't work in the main ./pages folder. When added on an api route or /api directly, or in an sub-folder of ./pages it runs.
> 
> e.g. ./pages/_middleware.ts wont work ./pages/app/_middleware.ts will work ./pages/api/_middleware.ts will work
> 
> Also in 12.0.5-canary.10

I have my middleware in sub folder but still not running
<img width="251" alt="Screenshot 2021-11-26 at 14 38 45" src="https://user-images.githubusercontent.com/32234038/143555765-c7d32574-65b5-43e0-86c5-1700e75116ec.png">

**Comment by huozhi** [comment]
Resolved with next 12.2
